x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+----
+
+[source,txt]
+----
+/kibana
+|_ README.txt
+|_ elasticsearch-ca.pem
+|_ sample-kibana.yml
+----
+--
+
+. If necessary, rename the `http.p12` file to match the name of your existing
+certificate for HTTP client communications. This keystore contains both a
+certificate and key.
+
+. Replace your existing keystore (such as `http.p12`) with the new keystore,
+ensuring that the file names match.
++
+IMPORTANT: If your keystore is password protected, the password
+is stored in the {es} keystore, _and_ the password needs to change, then you
+must update the `ES_PATH_CONF/elasticsearch.yml` file to use the filename of the
+new keystore file that you created.
+
+. Add the password for your private key to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Restart the node where you updated the keystore. {es} will reload the file and
+its configuration and start presenting the new  certificate to the incoming TLS
+connections.
+
+. (Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. On each additional node in your cluster, stop {es} and {kib}, update the
+keystore, and restart {es} and {kib}.
+
+[[update-node-certs-different]]
+=== Update security certificates with a different CA
+++++
+<titleabbrev>With a different CA</titleabbrev>
+++++
+If your CA is changing, you'll need to generate a new CA certificate and key.
+You can then add this CA certificate to your existing CA truststore so that all
+nodes in your cluster so know and trust it. Then, generate a new certificate for
+each node that is signed by the new CA certificate and instruct your nodes to
+use the new files. 
+
+[[node-certs-different-transport]]
+==== Generate a new certificate for the transport layer
+Create a new CA certificate and add it to your existing CA truststore. After
+{es} reloads the file, you can remove the old CA certificate from your
+truststore.
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+security certificates that are currently in use. You'll use the same names
+for your new certificates.
++
+In this example, the keystore and truststore are using different files.
+Your configuration might use the same file for both the keystore and the 
+truststore. Elastic recommends using separate files. 
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. On *any* node in your cluster, generate a new CA certificate. You only need
+to complete this step one time.
++
+NOTE: If your organization has its own CA, then use that CA to generate a new
```

Comment:
This is not accurate. If the organization has its own CA then what the users would need to do is to generate CSRs and send them to the CA to get back signed certificates

Reply:
We can change this to read:
>NOTE: If your organization has its own CA, you'll need to  <<certutil-csr,generate Certificate Signing Requests (CSRs)>>. CSRs contain information that your CA uses to generate and sign a security certificate instead of using self-signed certificates that the `elasticsearch-certutil` tool generates.